iml-gui/crate/src/page/filesystem.rs

```diff
@@ -1,10 +1,229 @@
+                T::td_view(plain!(f.mount_command.clone()))
+            ],
+        ])
+    ]
+}
+
+fn targets<I>(caption: &str, model: &Model, tgts: &[&Arc<Target<TargetConfParam>>]) -> Node<I> {
+    div![
+        class![
+            C.bg_white,
+            C.border,
+            C.border_b,
+            C.border_t,
+            C.mt_4
+            C.rounded_lg,
+            C.shadow,
+        ],
+        div![
+            class![C.flex, C.justify_between, C.px_6, C._mb_px, C.bg_gray_200],
+            h3![class![C.py_4, C.font_normal, C.text_lg], caption]
+        ],
+        T::wrapper_view(vec![
+            T::thead_view(vec![
+                T::th_left(plain!("Name")),
+                T::th_left(plain!("Volume")),
+                T::th_left(plain!("Primary Server")).merge_attrs(class![C.whitespace_no_wrap]),
+                T::th_left(plain!("Failover Server")).merge_attrs(class![C.whitespace_no_wrap]),
+                T::th_left(plain!("Started on")).merge_attrs(class![C.whitespace_no_wrap]),
+            ]),
+            tbody![tgts.iter().map(|t| tr![
+                T::td_view(vec![
+                    a![
+                        class![C.text_blue_500, C.hover__underline],
+                        attrs! {At::Href => Route::Target(RouteId::from(t.id)).to_href()},
+                        &t.name
+                    ],
+                    span![class![C.mx_1], lock_indicator::view(&model.locks, &***t)],
+                    alert_indicator(&model.records.active_alert, &***t, true, Placement::Top),
+                ])
+                .merge_attrs(class![C.whitespace_no_wrap]),
+                T::td_view(volume_link(&model.records, t)).merge_attrs(class![C.w_4]),
+                T::td_view(server_link(Some(&t.primary_server), &t.primary_server_name)),
+                T::td_view(server_link(t.failover_servers.first(), &t.failover_server_name)),
+                T::td_view(server_link(t.active_host.as_ref(), &t.active_host_name)),
+                td!["TBD"],
+            ])]
+        ])
+    ]
 }
 
-pub fn view(_model: &Model) -> impl View<Msg> {
-    seed::empty()
+pub(crate) fn status_view<I, E: ResourceUri + ToCompositeId>(model: &Model, x: &E) -> Node<I> {
+    span![
+        class![C.whitespace_no_wrap],
+        span![class![C.mx_1], lock_indicator::view(&model.locks, x)],
+        alert_indicator(&model.records.active_alert, x, false, Placement::Top)
+    ]
+}
+
+pub(crate) fn mgs<I>(tgts: &HashMap<u32, Arc<Target<TargetConfParam>>>, f: &Filesystem) -> Node<I> {
+    if let Some(t) = tgts
+        .values()
+        .find(|t| t.kind == TargetKind::Mgt && f.mgt == t.resource_uri)
+    {
+        server_link(Some(&t.primary_server), &t.primary_server_name)
+    } else {
+        plain!("N/A")
+    }
+}
+
+pub(crate) fn clients_view<I>(f: &Filesystem) -> Node<I> {
+    plain!(match f.client_count {
+        Some(c) => c.round().to_string(),
+        None => "N/A".to_string(),
```

Comment:
What's the difference between `plain!("N/A")` and `"N/A".to_string()`?

Reply:
huh, looks like an alias for a new text node, neat.